src/V3WidthCommit.cpp

```diff
@@ -100,9 +100,15 @@ class WidthCommitVisitor final : public VNVisitor {
         if (local || prot) {
             const auto refClassp = VN_CAST(m_modp, Class);
             const char* how = nullptr;
-            if (local && defClassp && refClassp != defClassp) {
+            // Inner nested classes can access `local` or `protected` members of their outer class
+            const auto nestedAccess = [refClassp](const AstClass*, const AstNode* memberp) {
+                return memberp == refClassp;
+            };
+            if (local && defClassp
+                && ((refClassp != defClassp) && !(defClassp->existsMember(nestedAccess)))) {
```

Comment:
Do you have a negative test for this - that is make sure outer can't access local inner?

Reply:
I've added it.